plugins/kafka/README.md

```diff
@@ -0,0 +1,87 @@
+# Kafka Plugin
+
+<img src="./src/assets/screenshot-1.png">
+
+## Setup
+
+1. Run:
+
+```bash
+yarn add @backstage/plugin-kafka @backstage/plugin-kafka-backend
+```
+
+2. Add the plugin backend:
+
+In a new file named `kafka.js` under `backend/src/plugins`:
+
+```js
+import { createRouter } from '@backstage/plugin-kafka-backend';
+import { PluginEnvironment } from '../types';
+
+export default async function createPlugin({
+  logger,
+  config,
+}: PluginEnvironment) {
+  return await createRouter({ logger, config });
+}
+```
+
+And then add to `packages/backend/src/index.ts`:
+
+```js
+async function main() {
+  // ...
+  const kafkaEnv = useHotMemoize(module, () => createEnv('kafka'));
+  // ...
+
+  const apiRouter = Router();
+  // ...
+  apiRouter.use('/kafka', await kafka(kafkaEnv));
+  // ...
+```
+
+3. Add the plugin frontend to `packages/app/src/plugin.ts`:
+
+```js
+export { plugin as Kafka } from '@backstage/plugin-kafka';
+```
+
+4. Register the plugin frontend router in `packages/app/src/components/catalog/EntityPage.tsx`:
+
+```jsx
+import { Router as KafkaRouter } from '@backstage/plugin-kafka';
+
+// Then, somewhere inside <EntityPageLayout>
+
+<EntityPageLayout.Content
+  path="/kafka/*"
+  title="Kafka"
+  element={<KafkaRouter entity={entity} />}
+/>;
+```
+
+5. Add broker configs for the backend in your `app-config.yaml`:
+
+```yaml
+kafka:
+  clientId: backstage
+  brokers:
+    - localhost:9092
+```
+
+6. Add `kafka.apache.org/consumer-group` annotation to your services:
+
+```yaml
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  # ...
+  annotations:
+    kafka.apache.org/consumer-group: consumer-group-name
```

Comment:
Looking at this annotation from an evolution point of view ... I'm not entirely sure it will be flexible for some setups and use cases. For one thing, it can be common to use multiple consumer groups within a single application to isolate unrelated topics, etc. Another thing that comes to mind is this setup is assuming a specific broker/cluster environment for all entities in the catalog (since it's configured in the `app-config.yaml`).

This brings up the question as to what some future use cases will be. I bring this up, because I think introducing a new annotation requires some deeper thought into how it will evolve and be leveraged for existing features. For instance how will topics be handled, different environments, schema registries, etc.  It's possible other `kafka.apache.org/...` annotations could be introduced, but it's worth thinking about this up front since changing annotations later becomes more disruptive then code changes.

Reply:
Interesting! I haven't thought about where I want to take this plugin yet, but basically along the lines that we discussed in #3791. But let's think about multiple clusters or multiple consumer groups since that's definitely something we'll need to support. How about:
1. Change this annotation to plural (so `kafka.apache.org/consumer-group`) and then allow users to specify a comma separated list of consumer groups for the service. I again prefer to do it in a separate PR to keep PRs simple but can already change the annotation ofc.
2. Change the broker / cluster definition in `app-config.yaml` to support multiple clusters. So something like this I guess:
```yaml
kafka:
  clientId: backstage
  clusters:
    - name: prod-main
      brokers:
         - ...
    - name: data
      brokers:
         - ...
```

Then, maybe change the annotation of the component to be something like `cluster:consumer-group-name`.

Reply:
I'm sure we'll end up having to have config like that, not sure with what exact fields though. Maybe some data can be fetched from the clusters themselves.

Reply:
I believe these are the minimal set of things we need to have configured so we'll be able to connect to the clusters and provide meaningful information on components.

Reply:
Might be worth making the `kafka.apache.org/consumer-group` plural right away as part of this PR. I suspect the value of the annotation can be extended to support multiple without introducing a breaking change. On that note, perhaps singular would be fine as is even if it is extended to support multiple later on.